var/spack/repos/builtin/packages/rpm/package.py

```diff
@@ -20,67 +19,76 @@ class Rpm(AutotoolsPackage):
+    variant('selinux', default=False, description="Enable support for SELinux")
+    variant('python', default=False, description="Build Python bindings to RPM library")
+    variant('lua', default=True, description='Build with lua support')
+    variant('zstd', default=False, description='Build with zstd suport')
+    variant('posix', default=False, description="Enable POSIX.1e draft 15 file capabilities support")
+    variant('gpg', default=False, description="Install gpg for using cryptographic signatures")
     variant('openmp', default=True, description="OpenMP multithreading support")
-    variant('docs', default=False, description='build documentation')
+    variant('nls', default=False, description='Enable native language support')
 
     # Always required
     depends_on('popt')
 
     # Without this file patch, we don't detect lua
-    depends_on('lua+pcfile@5.3.5:')
-
-    depends_on('autoconf', type='build')
-    depends_on('automake', type='build')
-    depends_on('libtool',  type='build')
+    depends_on('lua+pcfile@5.3.5:', when='+lua')
 
     # Enable POSIX.1e draft 15 file capabilities support
     depends_on('libcap', when="+posix")
-    depends_on('berkeley-db@4.5:')
+    depends_on('berkeley-db@4.5:', when='berkeley-db=full')
+    depends_on('berkeley-db@4.5:', when='berkeley-db=readonly')
 
-    # Required for National Language Support, if not present autopoint error
-    depends_on('gettext')
+    depends_on('gettext', when='+nls')
+    depends_on('gettext', type='build')
+    depends_on('iconv')
     depends_on('file')  # provides magic.h
     depends_on('libarchive')
 
-    # suppot for cryptographic signatures
-    depends_on('gnupg', when="+gpg")
+    # support for cryptographic signatures
+    depends_on('gnupg', when='+gpg')
 
     # cryptographic library to support digests and signatures
-    depends_on('libgcrypt', when='-openssl')
-    depends_on('openssl@1.0.2:', when='+openssl')
+    depends_on('libgcrypt', when='crypto=libgcrypt')
+    depends_on('openssl@1.0.2:', when='crypto=openssl')
 
     # RPM needs some database, ndb requires no extra dependencies but sqlite does
     depends_on('sqlite@3.22.0:', when='+sqlite')
 
     # Python 2.x support is being deprecated
     depends_on('python@3.1:', when='+sqlite')
 
-    # compression support
+    # compression support -- there is no configure option for many of these
+    # and they autodetect the libraries, so it's better to just make them
+    # hard requirements to avoid linking against system libraries.
     depends_on('zlib')
-
-    # Desired to install these formats for use
     depends_on('bzip2')
     depends_on('gzip')
     depends_on('xz')
+    depends_on('lzma')
+    depends_on('zstd', when='+zstd')
 
     # java jar dependency analysis (already requirement for lua)
     depends_on('unzip', type='run')
 
-    # Documentation dependencies
-    depends_on('doxygen', type="build", when="+docs")
+    # Build dependencies
```

Comment:
I'm happy to review but I don't think I am qualified to be a maintainer. For this line, why did you remove the +docs variant? Having docs build by default I don't think is what we'd want.

Reply:
Well, that was a useful review :p I couldn't turn off the docs, it seems to build is as part of `make all`

Reply:
Ah, it picks up `doxygen` in my PATH. And indeed they have been building docs for the past 14 years by default:

https://github.com/rpm-software-management/rpm/commit/39454472fc4ca6f776b44edc742f0a307a28ab53